iot_input/models/iot_device_input.py

```diff
@@ -21,6 +22,12 @@ class IotDeviceInput(models.Model):
         string='Language',
     )
 
+    _sql_constraints = [(
+        'serial_unique',
+        'unique(device_id, serial)',
```

Comment:
I understand your change, but I think serials should be unique by all the inputs, not only on the device, so I vote for `unique(serial)`. On current installed databases, if there is a problem with it, an error will be raised on the update, but the installation will continue, so, users should need to review all. (not really a problem for me)

Reply:
I think we should first establish what constitutes a serial. If I understand your point of view, you take it like a serial number that should be unique.
As I see it an input serial is just a unique addressable string specific to that device. I take it in terms of industrial PLC where you have a PLC unit (unique address device) and its inputs (referred to by Ix i.e. I1, I2, ....) and outputs (Qx i.e. Q1, Q2,....) that are unique to that specific device.
Come to think of it, maybe it would make sense to not use serial and create another 'address' field?

Reply:
Make sense for me, the migration should need to move the current serial to this address

Reply:
But in that context address should NOT be the same as serial.
I think there should be unique(serial) if set and unique(device_id.address, address) if set (both using @constrains to perform the checks).
Then maybe two endpoints should exist: One calling by serial (pretty much as the one in master) and a second one using device and input address.

Reply:
I know, but current databases are, so we need to solve this problem for them, isn't it?